### PR TITLE
Move `server upgrade` to `instance upgrade`

### PR DESCRIPTION
**Pull request by fantix**
Also added `--local-minor` and `--to-latest` explicit options, error
and help messages are also updated correspondingly.

### CODE REVIEW

File: src/server/upgrade.rs [hunk 1]
```diff
@@ -39,23 +40,55 @@ pub enum ToDo {
     InstanceUpgrade(String, Option<VersionQuery>),
 }
 
-fn interpret_options(options: &Upgrade) -> ToDo {
-    if let Some(name) = &options.name {
+fn interpret_options(options: &Upgrade) -> anyhow::Result<ToDo> {
+    if options.local_minor {
+        if options.name.is_some() {
+            Err(anyhow::anyhow!(
+                "Cannot execute minor version upgrade on a single instance"
```
Comment (elprans) [inline src/server/upgrade.rs:47]:
```suggestion
                "Cannot perform minor version upgrade on a single instance"
```

File: src/server/upgrade.rs [hunk 2]
```diff
@@ -39,23 +40,55 @@ pub enum ToDo {
...
+            )).hint(
+                "Run `edgedb instance upgrade --local-minor` without \
+                instance name.")?;
```
Comment (elprans) [inline src/server/upgrade.rs:50]:
```suggestion
                "Run `edgedb instance upgrade --local-minor` without \
                specifying an instance.")?;
```

**Comment by tailhook** [comment]
Looks good, I think two things are missing so far:
1. erroring out `instance upgrade` on instances linked to a project
2. minor version check on project upgrade

But feel free to submit that as follow up PR.

**Comment by fantix** [comment]
Yes, thanks! I'm creating follow-up PRs.

**Review by elprans — APPROVED** [review]

**Pull request merged**